etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/TransformationRuleEngine.java

```diff
@@ -36,23 +37,18 @@ public void unloadRules() {
 
     @Override
     public void ensureRulesLoaded() throws RuleLoaderException {
-        // Double-checked locking - needed to protect from excessive sync locks
-        if (rules.isEmpty()) {
-            synchronized (this) {
-                if (rules.isEmpty()) {
-                    InputStream resourceStream =
-                            getClass()
-                                    .getClassLoader()
-                                    .getResourceAsStream(ruleDefinitionsFileName);
-                    if (resourceStream == null) {
-                        throw new RuleLoaderException(
-                                "File not found: " + ruleDefinitionsFileName,
-                                new FileNotFoundException());
-                    }
-                    List<TransformationRule> parsedRules =
-                            ruleLoader.loadRules(resourceStream, new TypeReference<>() {});
-                    this.rules.addAll(parsedRules);
+        synchronized (rules) {
```

Comment:
I initially didn't want to remove the containing block, but my understanding from reading sync block params I thought that it would be fine if we kept the scope to the rules array.

I have an alternative solution using lock primitives that should also work. Will give that a shot.

Reply:
What was the issue that SonarCloud created when the initial `volatile` solution was used?

Reply:
Interesting.  I definitely need to read up on the `volatile` keyword to better understand what it's all about.

I see your current solution, and perhaps that's the right solution?  Looks like some unit tests failed, but maybe they can just be fixed?

Reply:
Yeah, I'll be making some changes locally

Reply:
It passes with the latest changes!